### PR TITLE
[SPARK-7608] Clean up old state in RDDOperationGraphListener

### PR DESCRIPTION
**Pull request by andrewor14**
This is necessary for streaming and long-running Spark applications. @zsxwing @tdas

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/ui/scope/RDDOperationGraphListener.scala [hunk 1]
```diff
@@ -58,7 +63,16 @@ private[ui] class RDDOperationGraphListener(conf: SparkConf) extends SparkListen
     }
     jobIdToStageIds(jobId) = stageInfos.map(_.stageId).sorted
 
-    // Remove graph metadata for old stages
+    // Remove state for old jobs
+    if (jobIds.size >= retainedJobs) {
```
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/ui/scope/RDDOperationGraphListener.scala]: You forget to insert `jobId` to `jobIds`. Could you write a simple test to verify we won't leak anything.

**Comment by tdas** [comment]
You have to add unit tests that check for cleanup.

**Comment by zsxwing** [comment]
LGTM

**Comment by zsxwing** [comment]
retest this please

**Comment by andrewor14** [comment]
retest this please

**Comment by andrewor14** [comment]
Master 1.4

**Pull request closed without merging**